src/payloadlogger.jl

```diff
@@ -2,7 +2,7 @@ using Logging: AbstractLogger, Logging
 
 struct ScopePayloadLogger <: AbstractLogger
     logger::AbstractLogger
-    scope::Scope
+    scope::Union{Scope, Nothing}
```

Comment:
this change was necessary to get `@with` working when the parent scope is `nothing`. However it independently resolves the error from #16. Probably there should be some tests for that though (but imo that's a separate thing)